src/Compilers/CSharp/Portable/Errors/LazyMissingNonNullTypesContextDiagnosticInfo.cs

```diff
@@ -10,32 +11,33 @@ namespace Microsoft.CodeAnalysis.CSharp
     internal sealed class LazyMissingNonNullTypesContextDiagnosticInfo : LazyDiagnosticInfo
```

Comment:
>LazyMissingNonNullTypesContextDiagnosticInfo [](start = 26, length = 44)

Feels like we don't need this anymore (we can do this check eagerly).

Reply:
> Feels like we don't need this anymore (we can do this check eagerly).

No, we cannot do this eagerly because this class is used when we cannot ask if a type is a value type due to the risk of going into a cycle.

---
In reply to: [239966087](https://github.com/dotnet/roslyn/pull/31607#discussion_r239966087) [](ancestors = 239966087)